rust/datafusion/src/optimizer/projection_push_down.rs

```diff
@@ -153,8 +160,8 @@ impl ProjectionPushDown {
                 // create the projected schema
                 let mut projected_fields: Vec<Field> =
                     Vec::with_capacity(projection.len());
-                for i in 0..projection.len() {
-                    projected_fields.push(schema.fields()[i].clone());
+                for i in &projection {
+                    projected_fields.push(table_schema.fields()[*i].clone());
```

Comment:
Do we need to remove duplicated fields here?

Reply:
I don't see any duplicated fields? This change is just to use the contents of the projection vector rather than the indices of the vector.